src/main/java/com/yelp/nrtsearch/server/luceneserver/IndexState.java

```diff
@@ -1251,20 +1253,22 @@ public synchronized MergeThread getMergeThread(
     return iwc;
   }
 
-  synchronized boolean getBooleanSetting(String name, boolean val) {
-    return settingsSaveState.get(name) == null ? val : settingsSaveState.get(name).getAsBoolean();
+  boolean getBooleanSetting(String name, boolean val) {
+    ThreadSafeJSONObject settings = saveState.getSettings();
+    return settings.get(name) == null ? val : settings.get(name).getAsBoolean();
```

Comment:
You will only want to get() the value once and use it twice, otherwise it may change between accesses.